research-software-directory/RSD-as-a-service
# 245 db limits and 292 concept DOI validation

# Define the limits for all VARCHAR columns in the database AND concept DOI validation button

Closes #245 
Closes #292 

Changes proposed in this pull request:
*  VARCHAR limits defined in all tables. Title, subtitle and description set to 200, 300 en 10.000 for both software and projects
*  TIMESTAMPTZ (time with timezone indication) applied
*  Forms using these columns show allowed max. length to the user
*  Find contributor, organisation or team member triggers after user types 2 characters
*  On edit software page, after DOI is provided, `validate DOI` button appears for user to confirm that it is valid concept DOI
*  If DOI is version DOI, like 10.5281/zenodo.6592693, and we can extract Concept DOI the button `UPDATE CONCEPT DOI` appears instead of Validate button.

How to test:
* `docker-compose down --volumes` remove old
* `docker-compose build`: build new
* `docker-compose up`: start
*  run data migrations script, it should not fail
*  create new software, try using values up to defined limits in all sections (info, contributors, organisations).
*  Add this DOI (version doi) and use validate DOI button, you should get warning message and the button update concept doi should appear
*  create new project, try using values up to defined limits in all sections (info, team, organisations).

## Example project input on "max"

![image](https://user-images.githubusercontent.com/9204081/173589908-33486685-ad26-46b4-bb7a-1a8b501159bd.png)

## Example validate concept DOI

![image](https://user-images.githubusercontent.com/9204081/173664884-9a48e354-65c6-46bf-8dd2-435303d4bb2f.png)

PR Checklist:
